operators/src/source/gdal_source/mod.rs

```diff
@@ -498,14 +549,22 @@ impl GdalRasterLoader {
             },
         );
 
-        if dataset_result.is_err() {
-            // TODO: check if Gdal error is actually file not found
+        if let Err(error) = &dataset_result {
+            let is_file_not_found = matches!(
+                error,
+                Error::Gdal {
+                    source: GdalError::NullPointer {
+                        method_name,
+                        msg
+                    },
+                } if *method_name == "GDALOpenEx" && (*msg == "HTTP response code: 404" || msg.ends_with("No such file or directory"))
+            );
```

Comment:
I guess it is wise to write tests for that behavior.

Reply:
3b007f8b0f53c02c061129542fef22082e452973